src/spikeinterface/postprocessing/tests/test_template_metrics.py

```diff
@@ -3,6 +3,36 @@
 import pytest
 
 
+def test_compute_new_template_metrics(small_sorting_analyzer):
+    """
+    Computes template metrics then computes a subset of quality metrics, and checks
+    that the old quality metrics are not deleted.
+
+    Then computes template metrics with new parameters and checks that old metrics
+    are deleted.
+    """
+
+    small_sorting_analyzer.compute("template_metrics")
```

Comment:
apologies for my ignorance, what does this call do vs. the one underneath? by default will this compute every template metric? and then `assert "half_width" in list(template_metric_extension.get_data().keys())` tests that they are not deleted, using `half_width` as an example? This all makes sense, however assuming I am correct above it the logic flow would be clearer by either a) only computing `half_width` in the first call or b) testing against all template metrics from the default template metrics name list in the assert.

Reply:
I've added some comments. Without any metric names, it will compute everything. With some names it will recompute those metrics. I was testing to make sure thing weren't being deleted. But yeah, it'd be better to test:
 1) _only_ half width is computed 
2) everything is computed 
3) nothing is deleted with half width is computed
4) everything is deleted when we turn `delete_existing_metric` on
Gimmie a sec...

Reply:
Done!